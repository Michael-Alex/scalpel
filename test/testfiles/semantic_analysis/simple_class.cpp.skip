class square
{
	public:
		square(const unsigned int size);

		unsigned int
		size() const;

		void
		size(const unsigned int size);

	private:
		unsigned int size_;
};

square::square(const unsigned int size):
	size_(size)
{
}

void
square::size(const unsigned int size)
{
	size_ = size;
}

unsigned int
square::size() const
{
	return size_;
}

