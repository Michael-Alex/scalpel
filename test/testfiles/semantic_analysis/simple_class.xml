<namespace id="n0">
 <class name="square" id="c0">
  <constructor access="public">
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualifiers const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualifiers>
     </type>
    </parameter>
   </parameters>
  </constructor>
  <destructor access="public" defined="false">
  </destructor>
  <simple_function name="size" access="public" const="true">
   <return_type>
    <fundamental_type type="unsigned int"/>
   </return_type>
  </simple_function>
  <simple_function name="size" access="public">
   <return_type>
    <fundamental_type type="void"/>
   </return_type>
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualifiers const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualifiers>
     </type>
    </parameter>
   </parameters>
  </simple_function>
  <variable name="size_" access="private">
   <type>
    <fundamental_type type="unsigned int"/>
   </type>
  </variable>
 </class>
</namespace>
