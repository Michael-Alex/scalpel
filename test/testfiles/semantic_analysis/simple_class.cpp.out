<namespace id="0">
 <class name="square" id="0">
  <constructor access="public">
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualified_type const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualified_type>
     </type>
    </parameter>
   </parameters>
  </constructor>
  <destructor access="public" defined="false">
  </destructor>
  <simple_member_function name="size" access="public" const="true">
   <return_type>
    <fundamental_type type="unsigned int"/>
   </return_type>
  </simple_member_function>
  <simple_member_function name="size" access="public">
   <return_type>
    <fundamental_type type="void"/>
   </return_type>
   <parameters>
    <parameter name="size">
     <type>
      <cv_qualified_type const="true">
       <fundamental_type type="unsigned int"/>
      </cv_qualified_type>
     </type>
    </parameter>
   </parameters>
  </simple_member_function>
  <member_variable name="size_" access="private">
   <type>
    <fundamental_type type="unsigned int"/>
   </type>
  </member_variable>
 </class>
</namespace>
